app/helpers/language_helper.py

```diff
@@ -45,7 +47,16 @@ def get_languages_context():
 def _get_language_context(language_code):
     return {
         "ISOCode": language_code,
-        "url": "?language_code=" + language_code,
+        "url": _get_query_with_language(language_code),
         "text": LANGUAGE_TEXT.get(language_code),
         "current": language_code == get_locale().language,
     }
+
+
+def _get_query_with_language(language_code):
+    query_mapping = parse_qs(urlparse(request.url).query)
+    query_mapping["language_code"] = [language_code]
+    query_string = "&".join(
+        f"{key}={value}" for key, values in query_mapping.items() for value in values
+    )
+    return f"?{query_string}"
```

Comment:
We should use flask's the already parsed`request.args`, and we should use `urlencode` from `urlparse` to encode the query string:
```suggestion
    request_args = {}
    for key, value in request.args.items():
        request_args[key] = value
    request_args["language_code"] = language_code
    return f"?{urlencode(request_args)}"
```

Reply:
Since it is a generator it can be:
```py
    request_args = dict(request.args.items())
    request_args["language_code"] = language_code
    return f"?{urlencode(request_args)}"
```